CONTRIBUTING.md

```diff
@@ -61,6 +61,51 @@ Pull requests should not be squash merged in order to keep complete commit messa
 This means that, when addressing review comments in order to keep the pull request in an always mergeable status, you will sometimes need to rewrite your branch's history and then force-push it with `git push --force-with-lease`.
 Useful git commands that can help a lot with this are `git commit --patch --amend` and `git rebase --interactive`. For more details consult the git man pages or online resources like [git-rebase.io](https://git-rebase.io/) or [The Pro Git Book](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).
 
+## Including patches
+
+While we do accept custom patches, we strongly encourage submitting patches
+upstream whenever possible. This practice means submitting the patches against
+the original software's repository, which has the potential to benefit the
+broader user community.
+
+For out-of-tree patches fetched using `fetchpatch`, avoid linking to GitHub pull
+requests, as these are mutable and can change with each new commit or
+force-push. Instead, use direct links to specific commit patches such as
+`https://github.com/owner/repo/pull/number/commits/sha.patch`. This ensures that
+the reference to your patch remains constant and accurate.
+
+For consistency and to streamline our patch management process, we recommend
+using `git format-patch` for formatting all patches. This will create a patch
+file that includes the commit message and the associated metadata along with the
+changes themselves. Remember, a well-structured patch is as critical as the code
+changes it contains.
+
+As part of our ongoing efforts to ensure the clarity, maintainability, and
+quality of our source code, we've implemented a policy that requires all
+contributors to provide explanatory comments with every patch submission.
+
+This means that each time you add a patch to the source code, it is mandatory to
+accompany it with a comment that clearly outlines its purpose, regardless of its
+filename.
```

Comment:
Also regardless of the description inside the patch itself?
When appropriate, I usually do `git format-patch` for patches that are only relevant for nixpkgs as well and then I add a descriptive commit message there as I do it with all other commits %)